src/Cameras/Inputs/freeCameraMouseWheelInput.ts

```diff
@@ -0,0 +1,402 @@
+     * controls.
+     * @param axis The axis to be moved. Set null to clear.
+     */
+    @serialize()
+    public set wheelXMoveScene(axis: Nullable<AXIS>) {
+        if (axis === null && this._wheelXAction !== this._rotateRelative) {
+            // Attempting to clear different _wheelXAction.
+            return;
+        }
+        this._wheelXAction = this._moveScene;
+        this._wheelXActionProperty = axis;
+    }
+
+    /**
+     * Get the configured movement axis (relative to the scene) the mouse wheel's
+     * X axis controls.
+     * @returns The configured axis or null if none.
+     */
+    public get wheelXMoveScene(): Nullable<AXIS> {
+        if (this._wheelXAction !== this._moveScene) {
+            return null;
+        }
+        return this._wheelXActionProperty;
+    }
+
+    /**
+     * Set which movement axis (relative to the scene) the mouse wheel's Y axis
+     * controls.
+     * @param axis The axis to be moved. Set null to clear.
+     */
+    @serialize()
+    public set wheelYMoveScene(axis: Nullable<AXIS>) {
+        if (axis === null && this._wheelYAction !== this._rotateRelative) {
+            // Attempting to clear different _wheelYAction.
+            return;
+        }
+        this._wheelYAction = this._moveScene;
+        this._wheelYActionProperty = axis;
+    }
+
+    /**
+     * Get the configured movement axis (relative to the scene) the mouse wheel's
+     * Y axis controls.
+     * @returns The configured axis or null if none.
+     */
+    public get wheelYMoveScene(): Nullable<AXIS> {
+        if (this._wheelYAction !== this._moveScene) {
+            return null;
+        }
+        return this._wheelYActionProperty;
+    }
+
+    /**
+     * Set which movement axis (relative to the scene) the mouse wheel's Z axis
+     * controls.
+     * @param axis The axis to be moved. Set null to clear.
+     */
+    @serialize()
+    public set wheelZMoveScene(axis: Nullable<AXIS>) {
+        if (axis === null && this._wheelZAction !== this._rotateRelative) {
+            // Attempting to clear different _wheelZAction.
+            return;
+        }
+        this._wheelZAction = this._moveScene;
+        this._wheelZActionProperty = axis;
+    }
+
+    /**
+     * Get the configured movement axis (relative to the scene) the mouse wheel's
+     * Z axis controls.
+     * @returns The configured axis or null if none.
+     */
+    public get wheelZMoveScene(): Nullable<AXIS> {
+        if (this._wheelZAction !== this._moveScene) {
+            return null;
+        }
+        return this._wheelZActionProperty;
+    }
+
+    /**
+     * A user configurable callback to be called on mouse wheel movement.
+     * To be used whenever the default functionality of this class does not
+     * change the required camera parameter by default.
+     */
+    @serialize()
+    public customCallback: Nullable<IFreeCameraMouseWheelCustomCallback> = null;
```

Comment:
Will refactor to use the Observer pattern instead.
Also, no reason the Observable couldn't be in the base class so any future implementations using the Base get it.